test/rulesets/LinearAlgebra/symmetric.jl

```diff
@@ -329,13 +329,7 @@
                 @testset for uplo in (:L, :U), hermout in (true, false)
                     A, ΔA = rand_matfun_input(f, TA, T, uplo, n, hermout), TA(randn(T, n, n), uplo)
                     Y = f(A)
-                    istypestable = try
-                        @inferred f(A)
-                        true
-                    catch ErrorException
-                        false
-                    end
-                    if istypestable
+                    if ChainRulesTestUtils._is_inferrable(f, A)
```

Comment:
I don't love that we are using an explictly nonexported function.
Do we even need to be testing `@inferred` here, or can we rely on `frule_test` to do that for us later

Reply:
Unfortunately we cannot use `frule_test` or `rrule_test` here because we need to work around the outputs of some functions being type-unstable. (and `rrule`'s won't agree with FiniteDifferences for `Symmetric`/`Hermitian`). I copied over a simplified `is_inferrable`.